### PR TITLE
Muriel colors: change code to new color API

### PR DESCRIPTION
**Pull request by nheagy**
Refs #11683

This changes most calls to the old color API (`WPStyleGuide.wordPressBlue()` to `.primary`, etc). Most changes are strictly according to the conversion table in the spreadsheet.

| Screen recording | Spreadsheet sample |
| ----- | ----- |
| color-change-to-new-api-sm | color table sample |

 I've made some changes where it's obvious what an appropriate semantic color would be, such as

```swift
likeButton.setTitleColor(WPStyleGuide.grey(), for: .normal)
// became
likeButton.setTitleColor(.textSubtle, for: .normal)
```

There's also a fun Objective-C API added in `UIColor+MurielColorsObjC.swift`.

To test:
- You can look through the app to see the effects of the color change, but screen-by-screen tweaking is for another PR
- It's key that this makes appropriate changes according to the spreadsheet. So look through the code and its 203 changed files?
- I'm sure you'll notice if something is off

Update release notes:

- [ ] If there are user facing changes, I have added an item to `RELEASE-NOTES.txt`.

### CODE REVIEW

File: WordPress/Classes/Models/DiffAbstractValue+Attributes.swift [hunk 1]
```diff
@@ -4,11 +4,11 @@ extension DiffAbstractValue {
         case .add:
             return [.backgroundColor: WPStyleGuide.extraLightBlue(),
```
Comment (frosty) [inline WordPress/Classes/Models/DiffAbstractValue+Attributes.swift]: Do we need to do these too?

File: WordPress/Classes/Models/DiffAbstractValue+Attributes.swift [hunk 2]
```diff
@@ -4,11 +4,11 @@ extension DiffAbstractValue {
...
                     .underlineStyle: NSNumber(value: 2),
-                    .underlineColor: WPStyleGuide.wordPressBlue()]
+                    .underlineColor: UIColor.primary]
         case .del:
             return [.backgroundColor: WPStyleGuide.extraLightRed(),
```
Comment (frosty) [inline WordPress/Classes/Models/DiffAbstractValue+Attributes.swift]: Do we need to do these too?

File: WordPress/Classes/ViewRelated/Post/Revisions/Style/WPStyleSuide+Revisions.swift [hunk 3]
```diff
@@ -1,3 +1,4 @@
+// TODO: Murielize these at their usepoints
```
Comment (frosty) [inline WordPress/Classes/ViewRelated/Post/Revisions/Style/WPStyleSuide+Revisions.swift]: Should be able to remove this file once we update the use points.

File: WordPress/Classes/System/WordPressAppDelegate.swift [hunk 4]
```diff
@@ -436,19 +436,19 @@ extension WordPressAppDelegate {
...
+        appearance.titleTextColor = .neutral(shade: .shade700)
         appearance.titleFont = WPStyleGuide.fontForTextStyle(.title2, fontWeight: .semibold)
 
-        appearance.bodyTextColor = WPStyleGuide.darkGrey()
+        appearance.bodyTextColor = .neutral(shade: .shade700)
         appearance.bodyFont = WPStyleGuide.fontForTextStyle(.body)
 
         appearance.actionFont = WPStyleGuide.fontForTextStyle(.headline)
         appearance.infoFont = WPStyleGuide.fontForTextStyle(.subheadline, fontWeight: .semibold)
         appearance.infoTintColor = WPStyleGuide.wordPressBlue()
```
Comment (frosty) [inline WordPress/Classes/System/WordPressAppDelegate.swift]: The app delegate still has quite a few references to old colors, for example here. Do a search for `WPStyleGuide.` in here (I know some are in feature-flagged non-Muriel code, but some aren't).

File: WordPress/Classes/ViewRelated/Activity/WPStyleGuide+Activity.swift [hunk 5]
```diff
@@ -72,13 +72,13 @@ extension WPStyleGuide {
         public static func getColorByActivityStatus(_ activity: Activity) -> UIColor {
             switch activity.status {
             case ActivityStatus.error:
-                return WPStyleGuide.errorRed()
+                return .error
             case ActivityStatus.success:
-                return WPStyleGuide.validGreen()
+                return .success
             case ActivityStatus.warning:
                 return WPStyleGuide.warningYellow()
```
Comment (frosty) [inline WordPress/Classes/ViewRelated/Activity/WPStyleGuide+Activity.swift]: `.warning`?

Comment (frosty) [reply WordPress/Classes/ViewRelated/Activity/WPStyleGuide+Activity.swift]: There are also other references to old colours higher up in this file.

File: WordPress/Classes/ViewRelated/Blog/NoResultsViewController.swift [hunk 6]
```diff
@@ -449,7 +449,8 @@ private extension NoResultsViewController {
     func configureButton() {
         actionButton.contentEdgeInsets = DefaultRenderMetrics.contentInsets
 
-        let normalImage = renderBackgroundImage(fill: WPStyleGuide.mediumBlue(), border: WPStyleGuide.wordPressBlue())
+        // TODO: Murielize these colors when buttons are done
+        let normalImage = renderBackgroundImage(fill: .primary(shade: .shade400), border: WPStyleGuide.wordPressBlue())
         let highlightedImage = renderBackgroundImage(fill: WPStyleGuide.wordPressBlue(), border: WPStyleGuide.wordPressBlue())
```
Comment (frosty) [inline WordPress/Classes/ViewRelated/Blog/NoResultsViewController.swift]: Can we not update this one, or are we waiting for something?

**Comment by frosty** [comment]
Also, a failing test needs fixing!

File: WordPress/Classes/Models/DiffAbstractValue+Attributes.swift (hunk 1 above)
Comment (nheagy) [reply WordPress/Classes/Models/DiffAbstractValue+Attributes.swift]: I had some trouble getting that screen to appear correctly, so I skipped it.

File: WordPress/Classes/ViewRelated/Post/Revisions/Style/WPStyleSuide+Revisions.swift (hunk 3 above)
Comment (nheagy) [reply WordPress/Classes/ViewRelated/Post/Revisions/Style/WPStyleSuide+Revisions.swift]: That's the motivation I needed!

**Comment by nheagy** [comment]
@frosty this is ready for another look! Thanks for catching those remaining uses of old colors.

If you want, you could checkout the new colors I picked for the diff view in revisions. I tried to get as close as I could to the existing colors.

image

**Pull request merged**